### ISSUE TITLE
Would like to see a working model of a sticky header and sticky footer that passes

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/wcag#1518]
These can be finicky with the CSS, getting the sticky header to smoothly scroll into place and then lock in place. I'm not sure that it can be done smoothly if padding added at a particular moment. For instance I think Boostrap sticky header and footer fail the current bullet.

Would like to see an implementation that "fixes" this so it passes the SC without being jumpy as the user scrolls.
https://getbootstrap.com/docs/4.0/components/navbar/

**Comment by alastc** [comment]
Hi David,

The CSS property I mentioned is [scroll-padding](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-padding), which I used on [this example](https://alastairc.uk/tests/wcag22-examples/sticky-footer4.html).

So on the HTML element (or whatever is scrolling) you'd add a scroll-padding of whatever the height of the sticky element is.

None of this impacts whether the sticky header is smooth or not, it impacts how much space the browser leaves as you tab through. Or any means of focus, it also solves the problem of following an in-page link which is then hidden by the header.

If you add scroll padding when there is no sticky header/footer, you'll just notice that the bottom link you are tabbing to is well above the bottom of the viewport.

If bootstrap adds that line of CSS, it has no impact on the 'smoothness' of the header, it just means the focus does not go behind it.

**Comment by alastc** [comment]
Hi @DavidMacDonald, does the answer above resolve this?

**Comment by alastc** [comment]
Hi @DavidMacDonald, 

I think this is an orthogonal issue to the focus-appearance SC. I.e. it is true that some are finicky, but the proposed solution for sticky headers/footers doesn't affect that.

I'm closing now due to no reply (and it's 'orthagonalness') but please re-open if I've missed something.
